### ISSUE TITLE
[OpenGL] Renders mode 4 second row twice, last row cut off

### ISSUE THREAD

**Issue by crawfxrd** [mgba-emu/mgba#1661]
mgba-qt renders the second row twice, and the last row is cut off from the display.

Doesn't happen in: 0.7.2 (c2b6692ce), 0.7.3 (7cd7a4165).
Happens in: f455650f0, 9886bce15, 0.8-b1 (13893803c), 0.8.0 (2368391f3), 7c0f3842a.

SDL:
![mgba-sdl](https://user-images.githubusercontent.com/5222085/74079455-f802c380-49f4-11ea-865c-20b8d64271b1.png)
Qt:
![mgba-qt](https://user-images.githubusercontent.com/5222085/74079457-fb964a80-49f4-11ea-9747-e43f1b1e4df4.png)

**Comment by endrift** [comment]
Which display driver?

**Comment by crawfxrd** [comment]
I'm on an X11 session using amdgpu.

**Comment by endrift** [comment]
No, I mean the Display driver setting in mGBA, under Audio/Video

**Comment by crawfxrd** [comment]
OpenGL driver. I just retested with Qt software rendering on 7c0f3842a and that works.

**Comment by endrift** [comment]
I can't reproduce. Can you track down when it started failing?

**Comment by crawfxrd** [comment]
Important piece of info for this is I'm drawing in mode 4.

I played with the settings and found both of these are required to produce the issue:

Audio/Video > Display driver: OpenGL
Enhancements > Video renderer: OpenGL

So going off that second settings, I went all the way back. It happens since 20ea552d1.
(Between 845041708 and that, drawing in mode 4 just shows a black screen.)

**Comment by endrift** [comment]
This has nothing to do with Qt then, and everything to do with the new OpenGL renderer that's not even in 0.7

**Comment by endrift** [comment]
I have a work in progress fix for this, however it causes the interpolation of the first few scanlines to become noisy so it won't be ready for 0.9.1.
